hypha/apply/stream_forms/forms.py

```diff
@@ -97,34 +99,47 @@ def clean(self):
                         email=self.data.get(field)
                     )
                     if is_registered:
                         user = get_user_by_email(email=email)
                         if not user:
-                            self.add_error(field, "Found multiple account")
+                            self.add_error(field, _("Found multiple account"))
                             raise ValidationError(
                                 mark_safe(
-                                    "Found multiple account for the same email. "
-                                    "Please login with the correct credentials or "
-                                    '<a href="mailto:{}">'
-                                    "contact to the support team"
-                                    "</a>.".format(settings.ORG_EMAIL)
+                                    _(
+                                        "Found multiple account for the same email. "
+                                        "Please login with the correct credentials or "
+                                        '<a href="mailto:{}">'
+                                        "contact to the support team"
+                                        "</a>."
+                                    ).format(settings.ORG_EMAIL)
                                 )
                             )
 
                         elif not user.is_active:
-                            self.add_error(field, "Found an inactive account")
+                            self.add_error(field, _("Found an inactive account"))
                             raise ValidationError(
                                 mark_safe(
-                                    "Found an inactive account for the same email. "
-                                    "Please use different email or "
-                                    '<a href="mailto:{}">'
-                                    "contact to the support team"
-                                    "</a>.".format(settings.ORG_EMAIL)
+                                    _(
+                                        "Found an inactive account for the same email. "
+                                        "Please use different email or "
+                                        '<a href="mailto:{}">'
+                                        "contact to the support team"
+                                        "</a>."
+                                    ).format(settings.ORG_EMAIL)
                                 )
                             )
+            elif value.label == ValueBlock._meta_class.label:
+                amount = self.data.get(field)
+                if amount:
+                    try:
+                        amount = float(amount)
+                        cleaned_data[field] = amount
+                    except ValueError:
+                        self.add_error(field, _("Invalid number"))
+                        pass
```

Comment:
Tested this and it add server side validation on top of the client side validation.

The default widget for `DecimalField` is `NumberInput` but since we want to set `min` I think we need to keep that line.

Reply:
Clean and simple so let go with this and we do the translation issues in a separate issue.

Reply:
Thanks @theskumar, I also just tested it. It is working great. I am updating the PR.